### PR TITLE
feat: don't round down in fromBaseUnit if displayDecimals are omitted

### PR DESCRIPTION
**Pull request by gomesalexandre**
## Description

Iterates on top of https://github.com/shapeshift/web/pull/4203, making sure we don't round down when unnecessary.

LLMgen'd description and tests

This pull request updates the fromBaseUnit() function to handle the displayDecimals parameter more safely and provides better control over the number of decimal places in the output. Previously, the function used the .decimalPlaces() method with the ROUND_DOWN mode, which could potentially cause loss of precision in certain cases.

The updated implementation of the fromBaseUnit() function checks if the displayDecimals parameter is provided and is a valid number. If it is, the function rounds down the result to the specified number of decimal places using the .decimalPlaces() method and then converts the resulting BigNumber to a string with exactly displayDecimals decimal places using the .toFixed(displayDecimals) method.

If the displayDecimals parameter is not provided or is not a valid number, the function will simply convert the full precision BigNumber result to a string using the .toFixed() method, without applying any rounding.

In addition to the updated implementation, this PR includes Jest tests to cover various input scenarios, ensuring that the fromBaseUnit() function behaves as expected for different input values and optional displayDecimals parameters. The test suite includes examples with lower and higher amounts of decimal places, such as 18-precision ETH amounts.

By implementing these changes, we improve the flexibility and safety of the fromBaseUnit() function, allowing for better control over the number of decimal places in the output while preserving the full precision when needed.

## Notice

- [x] Have you followed the guidelines in our [Contributing]('https://github.com/shapeshift/web/blob/develop/CONTRIBUTING.md) guide?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/shapeshift/web/pulls) for the same update/change?

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

N/A

## Risk

It's worth noting that this change is considered high-risk as the fromBaseUnit() function is used extensively throughout the codebase. Any changes to its behavior could potentially impact various parts of the application. To mitigate this risk, a comprehensive test suite has been added to ensure that the function behaves as expected in different scenarios. Additionally, thorough testing and review should be performed before deploying these changes to production.

tl;dr Test amounts across the app similar to https://github.com/shapeshift/web/pull/4203 and ensure Tx history,sends, receives, balances, swaps are showing correct amounts

## Testing

- Tx history, sends, receives, balances, swaps are displaying correct amounts
- DeFi opportunities display correct amounts
- Swapper is still happy

### Engineering

- Scrutinize I know how to write an if statement

### Operations

- ☝🏽

### CODE REVIEW

**Comment by gomesalexandre** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #4215** <a href="https://app.graphite.dev/github/pr/shapeshift/web/4215" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/shapeshift/web/4215?utm_source=stack-comment).

File: src/lib/math.ts [hunk 1]
```diff
@@ -4,17 +4,23 @@ import type { BN } from './bignumber/bignumber'
...
   precision: number,
   displayDecimals?: number,
 ): string => {
-  return bnOrZero(value)
-    .div(bn(10).pow(precision))
-    .decimalPlaces(displayDecimals ?? precision, BigNumber.ROUND_DOWN)
-    .toFixed()
+  const precisionNumber = bnOrZero(value).div(bn(10).pow(precision))
+
+  if (typeof displayDecimals === 'number') {
```
Comment (gomesalexandre) [inline src/lib/math.ts:17]: `.decimalPlaces(0)` is perfectly valid, so we can't use `if (displayDecimals)` to check for the argument being passed, since it would evaluate to false in case of 0
